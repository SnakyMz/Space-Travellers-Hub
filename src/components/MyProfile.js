import MyMission from './MyMission';
import MyRockets from './MyRockets';
import '../styles/MyProfile.css';

function MyProfile() {
  return (
    <section id="profile">
      <MyMission />
      <MyRockets />
    </section>
  );
}

export default MyProfile;
