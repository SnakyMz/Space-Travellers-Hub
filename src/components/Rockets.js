import { useSelector } from 'react-redux';

function Rockets() {
  const rockets = useSelector((store) => store.rockets);
  return (
    <section id="rockets">
      {rockets.map((rocket) => (
        <div className="rocket" key={rocket.id}>
          <img src={rocket.flickr_images} alt="Rocket" className="rocketImg" />
          <h3 className="rocketHead">{rocket.rocket_name}</h3>
          <p className="rocketDesc">{rocket.description}</p>
          <button type="button" className="Btn reserveBtn">Reserve Rocket</button>
        </div>
      ))}
    </section>
  );
}

export default Rockets;
