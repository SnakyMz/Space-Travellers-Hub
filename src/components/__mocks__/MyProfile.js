const planet = require('../../images/planet.png');

function MyProfile() {
  const missions = [
    {
      mission_id: 1,
      mission_name: 'Alpha',
      description: 'Mission to go to space',
      reserved: false,
    },
    {
      mission_id: 2,
      mission_name: 'Beta',
      description: 'Mission to go back to earth',
      reserved: true,
    },
  ];

  const rockets = [
    {
      id: 1,
      image: planet,
      name: 'Alpha',
      description: 'Rocket launched in space.',
      reserved: false,
    },
    {
      id: 2,
      image: planet,
      name: 'Beta',
      description: 'Rocket launched in space.',
      reserved: true,
    },
  ];
  const reservedRockets = rockets.filter((rocket) => rocket.reserved === true);

  return (
    <section id="profile">
      <section>
        <table>
          <caption><h2>My Missions</h2></caption>
          <tbody>
            {missions.map((mission) => {
              if (mission.reserved === true) {
                return (
                  <tr key={mission.mission_id}>
                    <td><h4>{mission.mission_name}</h4></td>
                  </tr>
                );
              }
              return null;
            })}
          </tbody>
        </table>
      </section>
      <section id="myRockets">
        <table>
          <caption><h2>My Rockets</h2></caption>
          <tbody>
            {reservedRockets.map((reservedRocket) => (
              <tr key={reservedRocket.id}><td><h4>{reservedRocket.name}</h4></td></tr>
            ))}
          </tbody>
        </table>
      </section>
    </section>
  );
}

export default MyProfile;
