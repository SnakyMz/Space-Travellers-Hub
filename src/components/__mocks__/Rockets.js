const planet = require('../../images/planet.png');

function Rockets() {
  const rockets = [
    {
      id: 1,
      image: planet,
      name: 'Alpha',
      description: 'Rocket launched in space.',
      reserved: false,
    },
    {
      id: 2,
      image: planet,
      name: 'Beta',
      description: 'Rocket launched in space.',
      reserved: true,
    },
  ];
  return (
    <section id="rockets">
      {rockets.map((rocket) => (
        <div className="rocket" id={rocket.id} key={rocket.id}>
          <img src={rocket.image} alt="Rocket" className="rocketImg" />
          <div className="rocketDetail">
            <h3 className="rocketHead">{rocket.name}</h3>
            <p className="rocketDesc">
              {rocket.reserved === true ? <span className="reserveBadge">Reserved</span> : ''}
              {rocket.description}
            </p>
            {rocket.reserved === false ? <button type="button" className="reserveRocket">Reserve Rocket</button> : <button type="button" className="cancelRocket">Cancel Reservation</button>}
          </div>
        </div>
      ))}
    </section>
  );
}

export default Rockets;
