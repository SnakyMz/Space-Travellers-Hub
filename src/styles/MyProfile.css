@media screen and (min-width: 768px) {
  #profile {
    display: grid;
    grid-template-columns: 1fr 1fr;
    gap: 5%;
  }
}
