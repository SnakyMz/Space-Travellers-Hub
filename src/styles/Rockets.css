#rockets {
  padding: 5%;
}

.rocket {
  padding-bottom: 5%;
}

.rocketImg {
  width: 100%;
}

.rocketDetail {
  padding: 1%;
}

.rocketHead {
  font-size: 1.5rem;
  padding-bottom: 1rem;
}

.rocketDesc {
  font-size: 1.1rem;
  padding-bottom: 1rem;
}

.reserveRocket {
  background-color: dodgerblue;
  color: white;
  font-size: 1rem;
  padding: 0.7rem 1rem;
  border: solid 0.1rem dodgerblue;
}

.reserveRocket:hover {
  background-color: white;
  color: dodgerblue;
}

.cancelRocket {
  color: slategray;
  border: solid 0.1rem slategray;
  font-size: 1rem;
  padding: 0.7rem 1rem;
}

.cancelRocket:hover {
  background-color: slategray;
  color: white;
}

@media screen and (min-width: 768px) {
  #rockets {
    padding: 2%;
  }

  .rocket {
    display: grid;
    grid-template-columns: 1fr 3fr;
    gap: 2%;
    padding-bottom: 2%;
  }
}
