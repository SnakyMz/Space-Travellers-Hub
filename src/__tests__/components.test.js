import { render } from '@testing-library/react';
import Rockets from '../components/Rockets.js';
import MyProfile from '../components/MyProfile.js';

jest.mock('../components/Rockets.js');
jest.mock('../components/MyProfile.js');

describe('Testing Rendering', () => {
  it('Should render Rockets section', () => {
    const rockets = render(<Rockets />);
    expect(rockets).toMatchSnapshot();
  });

  it('Should render MyProfile section', () => {
    const profile = render(<MyProfile />);
    expect(profile).toMatchSnapshot();
  });
});