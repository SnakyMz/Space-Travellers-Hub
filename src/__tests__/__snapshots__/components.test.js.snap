// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Rendering Should render Header section 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <div
          class="logo"
        >
          <img
            alt="hub-logo"
            src="planet.png"
          />
          <h2>
            Space Traveler's Hub
          </h2>
        </div>
        <nav>
          <a
            class="rocket-link"
            href="../Rockets.js"
          >
            Rockets
          </a>
          <a
            class="mission-link visit"
            href="../Missions.js"
          >
            Missions
          </a>
          <div
            class="divider"
          />
          <a
            class="profile-link"
            href="../MyProfile.js"
          >
            My Profile
          </a>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header>
      <div
        class="logo"
      >
        <img
          alt="hub-logo"
          src="planet.png"
        />
        <h2>
          Space Traveler's Hub
        </h2>
      </div>
      <nav>
        <a
          class="rocket-link"
          href="../Rockets.js"
        >
          Rockets
        </a>
        <a
          class="mission-link visit"
          href="../Missions.js"
        >
          Missions
        </a>
        <div
          class="divider"
        />
        <a
          class="profile-link"
          href="../MyProfile.js"
        >
          My Profile
        </a>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing Rendering Should render Missions section 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="missions-list"
      >
        <table
          class="missions"
        >
          <thead>
            <tr>
              <th>
                <h3>
                  Mission
                </h3>
              </th>
              <th>
                <h3>
                  Description
                </h3>
              </th>
              <th>
                <h3>
                  Status
                </h3>
              </th>
              <th>
                <small />
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                <h4>
                  Alpha
                </h4>
              </td>
              <td>
                Mission to go to space
              </td>
              <td>
                <small
                  class="not-member"
                >
                  NOT A MEMBER
                </small>
              </td>
              <td>
                <button
                  class="join"
                  type="button"
                >
                  Join Mission
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <h4>
                  Beta
                </h4>
              </td>
              <td>
                Mission to go back to earth
              </td>
              <td>
                <small
                  class="member"
                >
                  Active Member
                </small>
              </td>
              <td>
                <button
                  class="cancel"
                  type="button"
                >
                  Leave Mission
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="missions-list"
    >
      <table
        class="missions"
      >
        <thead>
          <tr>
            <th>
              <h3>
                Mission
              </h3>
            </th>
            <th>
              <h3>
                Description
              </h3>
            </th>
            <th>
              <h3>
                Status
              </h3>
            </th>
            <th>
              <small />
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              <h4>
                Alpha
              </h4>
            </td>
            <td>
              Mission to go to space
            </td>
            <td>
              <small
                class="not-member"
              >
                NOT A MEMBER
              </small>
            </td>
            <td>
              <button
                class="join"
                type="button"
              >
                Join Mission
              </button>
            </td>
          </tr>
          <tr>
            <td>
              <h4>
                Beta
              </h4>
            </td>
            <td>
              Mission to go back to earth
            </td>
            <td>
              <small
                class="member"
              >
                Active Member
              </small>
            </td>
            <td>
              <button
                class="cancel"
                type="button"
              >
                Leave Mission
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing Rendering Should render MyProfile section 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        id="profile"
      >
        <section>
          <table>
            <caption>
              <h2>
                My Missions
              </h2>
            </caption>
            <tbody>
              <tr>
                <td>
                  <h4>
                    Beta
                  </h4>
                </td>
              </tr>
            </tbody>
          </table>
        </section>
        <section
          id="myRockets"
        >
          <table>
            <caption>
              <h2>
                My Rockets
              </h2>
            </caption>
            <tbody>
              <tr>
                <td>
                  <h4>
                    Beta
                  </h4>
                </td>
              </tr>
            </tbody>
          </table>
        </section>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      id="profile"
    >
      <section>
        <table>
          <caption>
            <h2>
              My Missions
            </h2>
          </caption>
          <tbody>
            <tr>
              <td>
                <h4>
                  Beta
                </h4>
              </td>
            </tr>
          </tbody>
        </table>
      </section>
      <section
        id="myRockets"
      >
        <table>
          <caption>
            <h2>
              My Rockets
            </h2>
          </caption>
          <tbody>
            <tr>
              <td>
                <h4>
                  Beta
                </h4>
              </td>
            </tr>
          </tbody>
        </table>
      </section>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing Rendering Should render Rockets section 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        id="rockets"
      >
        <div
          class="rocket"
          id="1"
        >
          <img
            alt="Rocket"
            class="rocketImg"
            src="planet.png"
          />
          <div
            class="rocketDetail"
          >
            <h3
              class="rocketHead"
            >
              Alpha
            </h3>
            <p
              class="rocketDesc"
            >
              Rocket launched in space.
            </p>
            <button
              class="reserveRocket"
              type="button"
            >
              Reserve Rocket
            </button>
          </div>
        </div>
        <div
          class="rocket"
          id="2"
        >
          <img
            alt="Rocket"
            class="rocketImg"
            src="planet.png"
          />
          <div
            class="rocketDetail"
          >
            <h3
              class="rocketHead"
            >
              Beta
            </h3>
            <p
              class="rocketDesc"
            >
              <span
                class="reserveBadge"
              >
                Reserved
              </span>
              Rocket launched in space.
            </p>
            <button
              class="cancelRocket"
              type="button"
            >
              Cancel Reservation
            </button>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      id="rockets"
    >
      <div
        class="rocket"
        id="1"
      >
        <img
          alt="Rocket"
          class="rocketImg"
          src="planet.png"
        />
        <div
          class="rocketDetail"
        >
          <h3
            class="rocketHead"
          >
            Alpha
          </h3>
          <p
            class="rocketDesc"
          >
            Rocket launched in space.
          </p>
          <button
            class="reserveRocket"
            type="button"
          >
            Reserve Rocket
          </button>
        </div>
      </div>
      <div
        class="rocket"
        id="2"
      >
        <img
          alt="Rocket"
          class="rocketImg"
          src="planet.png"
        />
        <div
          class="rocketDetail"
        >
          <h3
            class="rocketHead"
          >
            Beta
          </h3>
          <p
            class="rocketDesc"
          >
            <span
              class="reserveBadge"
            >
              Reserved
            </span>
            Rocket launched in space.
          </p>
          <button
            class="cancelRocket"
            type="button"
          >
            Cancel Reservation
          </button>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
